### PR TITLE
Adds Headphones!

### PR DESCRIPTION
**Pull request by silicons**
Fits on either neck, head, or ear slots.
Action button to turn on/off
kekkles

:cl:
rscadd: Headphones have been provided to the station in Autodrobes, mixed wardrobes, and fitness wardrobes. Please use them responsibly, and remember to focus on your job above all else!
experimental: Headphones fit in head, ears, OR neck slots!
/:cl:

### CODE REVIEW

**Comment by kevinz000** [comment]
FUGG XD

**Comment by Supermichael777** [comment]
make it play an instrument just to you!

**Comment by kevinz000** [comment]
@Supermichael777 remind me in a later pr but before the freeze :P

File: code/modules/clothing/ears/ears.dm [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	name = "earmuffs"
+	desc = "Protects your hearing from loud noises, and quiet ones as well."
+	icon_state = "earmuffs"
+	item_state = "earmuffs"
+	strip_delay = 15
+	equip_delay_other = 25
+	resistance_flags = FLAMMABLE
+
+/obj/item/clothing/ears/earmuffs/Initialize(mapload)
+	..()
```
Comment (lzimann) [inline code/modules/clothing/ears/ears.dm]: `. = ..()`

File: code/modules/clothing/ears/ears.dm [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	actions_types = list(/datum/action/item_action/toggle_headphones)
+	var/icon_state_on = "headphones_on"
+	var/icon_state_off = "headphones_off"
+	var/item_state_on = "headphones_on"
+	var/item_state_off = "headphones_off"
+	var/headphones_on = FALSE
+
+/obj/item/clothing/ears/headphones/ui_action_click(owner, action)
+	headphones_on = !headphones_on
+	if(headphones_on)
```
Comment (lzimann) [inline code/modules/clothing/ears/ears.dm]: Change the initial icon/item state to `headphones`, then you can do:
`item_state = "[initial(icon_state)]_[headphones_on ? "on" : "off"]"` for both icon and item states. And then you don't have to declare 4 new variables for this.
Also move the icon operations to `update_icon` and call it on initialize

**Comment by XDTM** [comment]
They should make you deaf while active, people who want to talk to you should grab your attention so you take them off to listen

**Comment by kevinz000** [comment]
@XDTM fluff item, i'd rather not unbalance it unless people really want me to do that :P

**Comment by WJohn** [comment]
I'll be respriting these soon enough.

**Comment by DoctoreMad** [comment]
>Fluff item

Geez how boring
How will I pretend I can't hear the janitor screaming at me for the floor buffer if the headphones actually do nothing on the code side?

File: code/game/objects/structures/crates_lockers/closets/fitness.dm [hunk 3]
```diff
@@ -11,6 +11,8 @@
 	new /obj/item/clothing/under/shorts/red(src)
 	new /obj/item/clothing/under/shorts/blue(src)
 	new /obj/item/clothing/under/shorts/green(src)
+	for(var/i in 1 to 4)
+		new /obj/item/clothing/ears/headphones(src)
```
Comment (lzimann) [inline code/game/objects/structures/crates_lockers/closets/fitness.dm]: They really should only spawn in one place instead of 3 different locations imo.

Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/structures/crates_lockers/closets/fitness.dm]: why put them in the miku locker

Comment (GunHog) [inline code/game/objects/structures/crates_lockers/closets/fitness.dm]: Shark girl needs her jams!

**Comment by improvedname** [comment]
>porting from paradise

**Comment by Alek2ander** [comment]
It should make you have trouble hearing people talk (apply the telecoms processor filter?), but not protect against loud noises

**Comment by improvedname** [comment]
Inb4 scientists wearing headphones 24/7 just to be anti social

**Comment by improvedname** [comment]
Also kevinz thats not how you wear headphones!
Your ears are up there and not how your wearing it currently!

**Comment by kevinz000** [comment]
can we balance memes later rather than now :^)

**Comment by Armhulen** [comment]
Make everyone else's font tiny when you have the headset

**Comment by Tacolizard** [comment]
@improvedname it's kevinz. he's from citadel. of course he's porting from paradise.

**Comment by Tacolizard** [comment]
it should also force you to yell when you wear them.

**Comment by kevinz000** [comment]
lmao
anyways i'd rather it be merged as a pure accessory item than something that can block sound
(mainly because I don't know how to askerisk text while it's on you, as I do believe it's either "block sound completely" or "let everything through", so i'd have to put more effort into it and i can't because feature freeze soon so i have to rush my prs right?)

**Comment by GunHog** [comment]
How about some more feature creep for you, Kev? You know the music that Robustin's dance machine plays? You could make it play songs from that!

**Comment by DoctoreMad** [comment]
You mean a actual feature freeze is coming or something?

**Comment by Kraseo** [comment]
real catpeople only have one set of ears and that's the ones atop their head, any person with cat ears and human ears is just a cosplayer

**Comment by dannno** [comment]
the animation is too cartoony tbqh

**Comment by octareenroon91** [comment]
Hey, is this supposed to be only in the theater vendor and not the public clothing vendor?

**Comment by kevinz000** [comment]
@DoctoreMad yes
@dannno already asked wjohn for other animations, i don't think this is that bad but he can do better
@octareenroon91 no?

File: code/game/machinery/vending.dm [hunk 4]
```diff
@@ -1051,7 +1051,8 @@ IF YOU MODIFY THE PRODUCTS LIST OF A MACHINE, MAKE SURE TO UPDATE ITS RESUPPLY C
 					/obj/item/clothing/mask/joy = 1,
 					/obj/item/clothing/head/cueball = 1,
 					/obj/item/clothing/under/scratch = 1,
-        			/obj/item/clothing/under/sailor = 1)
+        			/obj/item/clothing/under/sailor = 1,
+        			/obj/item/clothing/ears/headphones = 8)
```
Comment (octareenroon91) [inline code/game/machinery/vending.dm]: Autodrobe is theater access.
Clothesmate is public access. Starts around line 1150.

Comment (kevinz000) [inline code/game/machinery/vending.dm]: fugg XD

**Comment by kevinz000** [comment]
Changed numbers to 2 in autodrobe for clown/mime and 10 in general access for crew, sounds good?

File: code/datums/action.dm [hunk 5]
```diff
@@ -207,6 +207,15 @@
...
 	name = "Toggle Helmet Light"
 
+/datum/action/item_action/toggle_headphones
+	name = "Toggle Headphones"
+	desc = "UNTZ UNTZ UNTZ"
+
+/datum/action/item_action/toggle_headphones/Trigger()
+	var/obj/item/clothing/ears/headphones/H = target
+	if(istype(H))
+		H.toggle(owner)
```
Comment (Cyberboss) [inline code/datums/action.dm]: Why is this check necessary?

**Comment by Niknakflak** [comment]
If you're gonna make a PR showing off headphones, don't be a special snowflake and still have your dumb cat ears, you can barely see the sprite through that shit.  Kevinz....

File: code/datums/action.dm (hunk 5 above)
Comment (kevinz000) [inline code/datums/action.dm]: incase fuckups/badminnery happens.

**Comment by kevinz000** [comment]
nyaa

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**